controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -251,7 +251,18 @@ func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, cluster *
 		return r.upgradeControlPlane(ctx, cluster, kcp, ownedMachines, requireUpgrade)
 	}
 
-	// If we've made it this far, we we can assume that all ownedMachines are up to date
+	// If we've made it this far, we can assume that all ownedMachines are up to date
+
+	// Update kube-proxy image name
+	workloadCluster, err := r.managementCluster.GetWorkloadCluster(ctx, util.ObjectKey(cluster))
+	if err != nil {
+		// Just log and continue, control-plane cluster may be uninitialized
+		logger.Error(err, "failed to get remote client for workload cluster")
```

Comment:
Would it make sense to move this down after the switch?

Reply:
I guess it will just take longer to update kube-proxy. After the first replica is created as we will also wait to scale up before updating kube-proxy image. I moved this for now after the switch.

Reply:
That should be ok I think, the call to `upgradeControlPlane` has `scaleDownControlPlane` at the end, so we were technically already waiting for the scale-down to happen before updating kube-proxy, which is also what I was expecting.

The reason I suggested to move this down is that before/during initialization we could avoid logging often.

Reply:
Makes sense.